helpers/legacyStore.js

```diff
@@ -6,13 +6,31 @@ var assign = require("can-util/js/assign/assign");
 
 module.exports = function (count, make, filter) {
 	/*jshint eqeqeq:false */
-	var getUniqueId = (function () {
+	var nextUniqueId = (function () {
 		var i = 0;
 		return function () {
 			return i++;
 		}
 	})();
 
+	// Check to see if an existing item has the ID we want to assign
+	var isIdUnique = function (items, id) {
+		for (var key in items) {
+			if (items[key].id === id) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	var getUniqueId = function (items) {
+		var id = nextUniqueId();
```

Comment:
It is O(n^2). We can make it O(n) if we make a point-of-call cache. Since we don't copy the individual `items` if they are passed directly any item's ID can be changed throughout a test so non-point-of-call caching is ineffective. If you want to change `nextUniqueId()`'s `i=0` to say `i=100` to not bump into someone's items, that's fine as well. In test scenarios, the uniqueness guarantee (correctness) is more important than the minor amortized cost (performance).